reconcile/openshift_upgrade_watcher.py

```diff
@@ -72,13 +86,19 @@ def notify_upgrades_start(
                 + f"cluster `{cluster}` is currently "
                 + f"being upgraded to version `{version}`"
             )
-            handle_slack_notification(msg, slack, state, state_key, None)
+            handle_slack_notification(
+                msg=msg, slack=slack, state=state, state_key=state_key, state_value=None
+            )
 
 
-def notify_upgrades_done(clusters: list[dict], state: State, slack: Optional[SlackApi]):
+def notify_upgrades_done(
+    clusters: Iterable[ClusterV1], state: State, slack: Optional[SlackApi]
+) -> None:
     for cluster in clusters:
-        cluster_name = cluster["name"]
-        version = cluster["spec"]["version"]
+        cluster_name = cluster.name
+        if not cluster.spec:
+            raise RuntimeError(f"Cluster '{cluster_name}' does not have any spec.")
+        version = cluster.spec.version
```

Comment:
Remove `cluster_name` and use 'cluster.name' instead (same for `version`). This will be much easier to read :)

Reply:
Good point. Fixed :)